food-truck-frontend/src/components/notifications/NotificationWatcherComponent.tsx

```diff
@@ -0,0 +1,45 @@
+import React, {useEffect, useState} from 'react';
+import {NotificationListProps} from "./NotificationListComponent";
+import getUserInfo from "../../util/token";
+
+function NotificationWatcherComponent(props: NotificationListProps) {
+  const [initialized, setInitialized]: [boolean, any] = useState(false);
+  const [notified, setNotified]: [boolean, any] = useState(false);
+  const notifyUnread = () => {
+    const foundItem = props.data.notifications.find((notification) => !notification.read);
+    if (foundItem !== null && foundItem !== undefined) {
+      setNotified(true);
+      alert('You have unread notification(s)');
+    }
+  };
+  useEffect(() => {
+    if (!initialized) {
+      // Load
+      var id: number | undefined = getUserInfo()?.userID;
+      if (id !== undefined && id !== 0) {
+        props.loadNotificationsFromBackend().then();
+      }
+      if (!notified) {
+        notifyUnread();
+      }
+      setInitialized(true);
+    } else {
+      const timer = setInterval(() => {
```

Comment:
Why is this doubly-nested? It looks like a carbon copy of the enclosing, repeating effect, so shouldn't it not add anything? Or am I misreading it completely?

Reply:
I refactored it out into a separate arrow function, if you were referring to the duplication.